include/netlink/route/neighbour.h

```diff
@@ -55,6 +58,12 @@ extern void			rtnl_neigh_unset_flags(struct rtnl_neigh *,
 						       unsigned int);
 extern unsigned int		rtnl_neigh_get_flags(struct rtnl_neigh *);
 
+extern void			rtnl_neigh_set_ext_flags(struct rtnl_neigh *,
+							 uint32_t);
+extern void			rtnl_neigh_unset_ext_flags(struct rtnl_neigh *,
+							   uint32_t);
+extern uint32_t			rtnl_neigh_get_ext_flags(struct rtnl_neigh *);
```

Comment:
I think this should be
```C
extern int rtnl_neigh_get_ext_flags(struct rtnl_neigh *neigh, uint32_t *flags);
```

(and return `-NLE_NOATTR` if the flags are not present).


That would allow to see, whether kernel sent those flags.

Indeed, kernel does not send `NEIGH_ATTR_EXT_FLAGS` if they are all zero. That seems unfortunate, because then a user cannot see whether the attribute is just not supported or whether it happens to be zero. That also lessens the use of knowing in `rtnl_neigh_get_ext_flags()` whether the flags are actually set.

Still, I think this should be done. Even if it's inconsistent with `rtnl_neigh_get_flags()`. But for `rtnl_neigh_get_flags()` the flags come from `nm->ndm_flags`, so it would always be present even with older kernel versions.

Reply:
I can agree with this, although I do see the merits in keeping it consistent with `get_flags` since the absence of the attribute doesn't seem to allow the user to make any meaningful conclusion, and they would need to do more error checking.

> Indeed, kernel does not send NEIGH_ATTR_EXT_FLAGS if they are all zero

It seems the kernel may send ext flags even if they are all zero if dumping FDB entries (see `fdb_fill_info`).

I've added a commit to make the function error if the flags aren't present.